File: new_code/1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(arr, target) {
    map = new Map ();
    for(let i=0;i<arr.length;i++){
        let diff = target-arr[i];
        if(map.has(diff)){
            return [map.get(diff), i];
        }
        else{
            map.set(arr[i], i);
        }
    }
};

File: new_code/1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] arr, int target) {

        Map<Integer, Integer> map = new HashMap<>();
        int i2 = 0;
        for(int i=0;i<arr.length;i++){
            map.put(arr[i], i);
        }
        for(int i=0;i<arr.length;i++){
            int diff = target-arr[i];
            if(map.containsKey(diff) && map.get(diff)!=i){
                return new int[]{i, map.get(diff)};
            }
        }
        return new int[]{};
       


    }
}
// done it by own 